cmd/editorconfig-checker/main.go

```diff
@@ -127,7 +127,9 @@ func main() {
 	errorCount := len(formattedErrors)
 	if errorCount != 0 {
 		config.Logger.PrintLogMessages(formattedErrors)
-		config.Logger.Error(fmt.Sprintf("\n%d errors found", errorCount))
+		if config.Format != "codeclimate" {
```

Comment:
I would like to suggest adding a constant for code climate

This way code that relies on this value could be easily identified

Reply:
I would take it on me to split `FormatErrors` into multiple independent functions once this PR is merged. Now with a fourth format supported this evergrowing function is too hard to read.

I also wondered if there was a neat way to represent the collection of possible output formats, so that the argument parsing code can discover the possible formats without us having to every time adjust it by hand.

Reply:
I would love that

It would help to have a type and a method to list all the formats.

The fact this code has to be maintained is q bit a pain

```go
        flag.StringVar(&c.Format, "format", "default", "specify the output format: default, gcc, github-actions, codeclimate")
```

Reply:
Opened #374 to remind me of the refactoring. 

Reading up on golang, I somehow did not find any mention how to implement essentially an enum type. @ccoVeille do you have seen anything alike in the past?